runtime/src/accounts_db.rs

```diff
@@ -5262,257 +5278,47 @@ impl AccountsDb {
     // Requires all stores in the slot to be re-written otherwise the accounts_index
     // store ref count could become incorrect.
     fn do_shrink_slot_v1(&self, slot: Slot, forced: bool) -> usize {
```

Comment:
Without going through this line by line is the idea here that this function is mostly the same code  as `do_shrink_slot_stores `?
@carllin had mentioned he thought it was time for us to get rid of the non cache code path and he was planning to do that at some point.

Reply:
Yea, should be the same code.

Reply:
I made a separate PR for that one: https://github.com/solana-labs/solana/pull/17385

Once that is merged I'll rebase to remove the diff. If you want to just look at the top 2 commits here, that should be indicative of the optimization changes.